packages/console/src/pages/ApplicationDetails/ApplicationDetailsContent/EndpointsAndCredentials/use-secret-table-columns.tsx

```diff
@@ -0,0 +1,100 @@
+import { type ApplicationSecret } from '@logto/schemas';
+import { compareDesc } from 'date-fns';
+import { useMemo } from 'react';
+import { useTranslation } from 'react-i18next';
+
+import ActionsButton from '@/components/ActionsButton';
+import CopyToClipboard from '@/ds-components/CopyToClipboard';
+import { type Column } from '@/ds-components/Table/types';
+import { Tooltip } from '@/ds-components/Tip';
+import useApi from '@/hooks/use-api';
+
+import styles from './index.module.scss';
+
+export type ApplicationSecretRow = Pick<ApplicationSecret, 'name' | 'value' | 'expiresAt'> & {
+  isLegacy?: boolean;
+};
+
+function Expired({ expiresAt }: { readonly expiresAt: Date }) {
+  const { t } = useTranslation(undefined, { keyPrefix: 'admin_console' });
+  return (
+    <Tooltip
+      content={t('application_details.secrets.expired_tooltip', {
+        date: expiresAt.toLocaleString(),
+      })}
+    >
+      <span className={styles.expired}>{t('application_details.secrets.expired')}</span>
+    </Tooltip>
+  );
+}
+
+type UseSecretTableColumns = {
+  appId: string;
+  onUpdated: (isLegacy: boolean) => void;
+};
+
+export const useSecretTableColumns = ({ appId, onUpdated }: UseSecretTableColumns) => {
+  const { t } = useTranslation(undefined, { keyPrefix: 'admin_console' });
+  const api = useApi();
+  const tableColumns: Array<Column<ApplicationSecretRow>> = useMemo(
+    () => [
+      {
+        title: t('general.name'),
+        dataIndex: 'name',
+        colSpan: 3,
+        render: ({ name }) => <span>{name}</span>,
+      },
+      {
+        title: t('application_details.secrets.value'),
+        dataIndex: 'value',
+        colSpan: 6,
+        render: ({ value }) => (
+          <CopyToClipboard
+            hasVisibilityToggle
+            displayType="block"
+            value={value}
+            className={styles.copyToClipboard}
+            variant="text"
+          />
+        ),
+      },
+      {
+        title: t('application_details.secrets.expires_at'),
+        dataIndex: 'expiresAt',
+        colSpan: 3,
+        render: ({ expiresAt }) => (
+          <span>
+            {expiresAt ? (
+              compareDesc(expiresAt, new Date()) === 1 ? (
+                <Expired expiresAt={new Date(expiresAt)} />
+              ) : (
+                new Date(expiresAt).toLocaleString()
```

Comment:
We have a component `DateTime` to render the time.

Reply:
discussed offline, refactor that component